### PR TITLE
hyperparamter tuning

### PR DESCRIPTION
**Pull request by tongh18**
fixed the passing dictionary

### CODE REVIEW

**Review by PSSF23 — CHANGES_REQUESTED** [review]
`black` format the files

**Review by PSSF23 — CHANGES_REQUESTED** [review]
You need to have better parameter file names/locations & resolve the current validation set problem. Also, uncomment the code you didn't use in testing.

**Review by PSSF23 — CHANGES_REQUESTED** [review]
Again, I believe having another `tune_cnn32` function is more reasonable & easier to integrate.

**Review by PSSF23 — CHANGES_REQUESTED** [review]
Leave most of `run_XXX` functions alone. It should directly use the best parameters output by `tune_XXX`.

@jzheng17 is doing tuning on RF. @tongh18 you should coordinate with her, and I want you two to have similar code structures.

**Review by PSSF23 — CHANGES_REQUESTED** [review]
Remember to revert everything related to development testing.

**Review by PSSF23 — CHANGES_REQUESTED** [review]
Also, the current tuning function:

- [x] doesn't tell me anything about the class numbers when saving parameters;
- [ ] only works for 1-layer CNN (so it should be `tune_XXX`);
- [x] doesn't correspond to naming conventions of existing files;
- [ ] has unnecessary things like rounding (no ranges are given so useless).

Please fix the above problems.

**Review by PSSF23 — CHANGES_REQUESTED** [review]
Your code wrongly saves results for each sample size & rewrites the result files many times. Correct it.

**Comment by PSSF23** [comment]
> Remember to revert everything related to development testing.

Still an issue.

**Review by PSSF23 — CHANGES_REQUESTED** [review]
- [ ] Your logic is reversed. You need to call `tune` within `run`, not the other way. And you defined all the loaders again in `run`, ruining the splitting of valid & test. If you do not keep consistent loaders the whole splitting is worthless. You should have the 3 sets in the beginning & use copies to create loaders. 
- [ ] Tuning time needs to be one-to-one corresponding to training time/test time. So for each sample size & for each class combination, not total.
- [ ] `test_dn_image_es_multiple` is redundant. Incorporate changes into `test_dn_image_es`.
- [ ] Dump dictionary of lists (so only dump once), not list of dictionaries and no line breaks. Your current code could not produce a readable file.
